src/components/app/details/triggerView/TriggerView.tsx

```diff
@@ -113,6 +116,82 @@ class TriggerView extends Component<TriggerViewProps, TriggerViewState> {
             })
         }
     }
+
+    fetchCommitHistoryFromDB(ciPipelineMaterialId: number, commitHash: string,workflows: WorkflowType[], _selectedMaterial: CIMaterialType){
+      fetchGitMaterialByCommitHash(ciPipelineMaterialId.toString(), commitHash)
+          .then((response) => {
+              const _result = response.result
+              if (_result) {
+                  _selectedMaterial.history = [
+                      {
+                          commitURL: _selectedMaterial.gitURL
+                              ? createGitCommitUrl(_selectedMaterial.gitURL, _result.Commit)
+                              : '',
+                          commit: _result.Commit || '',
+                          author: _result.Author || '',
+                          date: _result.Date ? ISTTimeModal(_result.Date, false) : '',
+                          message: _result.Message || '',
+                          changes: _result.Changes || [],
+                          showChanges: true,
+                          webhookData: _result.WebhookData,
+                          isSelected: true,
+                      },
+                  ]
+                  _selectedMaterial.isMaterialLoading = false
+              } else {
+                  _selectedMaterial.history = []
+                  _selectedMaterial.noSearchResultsMsg = `Commit not found for ‘${commitHash}’ in branch ‘${_selectedMaterial.value}’`
+                  _selectedMaterial.noSearchResult = true
+                  _selectedMaterial.isMaterialLoading = false
+              }
+              this.setState({
+                  workflows: workflows,
+              })
+          })
+          .catch((error: ServerErrors) => {
+              showError(error)
+              _selectedMaterial.isMaterialLoading = false
+              this.setState({
+                  workflows: workflows,
+              })
+          })
+    }
+
+    fetchMaterialByCommit(ciNodeId: number, pipelineName: string, ciPipelineMaterialId: number, commitHash = null) {
+      if (commitHash) {
+          let state = { ...this.state }
```

Comment:
Can we skip this and use this.state directly?

Reply:
Done